universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java

```diff
@@ -53,23 +64,40 @@ public Content render(HtmlBundle bundle) {
   public Content renderWithNav(
       Http.Request request, String userName, Messages messages, HtmlBundle bundle) {
     // TODO: This will set the html lang attribute to the requested language when we actually want
-    //  the rendered language.
-    Optional<Http.Cookie> language = request.cookies().get("PLAY_LANG");
+    // the rendered language.
+    Optional<String> language = languageUtils.getPreferredLangage(request);
     if (language.isPresent()) {
-      bundle.setLanguage(language.get().value());
+      bundle.setLanguage(language.get());
     }
-
     bundle.addHeaderContent(renderNavBar(request, userName, messages));
     return render(bundle);
   }
 
   private ContainerTag renderNavBar(Http.Request request, String userName, Messages messages) {
     Optional<UatProfile> profile = profileUtils.currentUserProfile(request);
-    return renderNavBar(profile, messages, userName);
-  }
 
-  private ContainerTag renderNavBar(
-      Optional<UatProfile> profile, Messages messages, String userName) {
+    String csrfToken = CSRF.getToken(request.asScala()).value();
+    Tag csrfInput = input().isHidden().withValue(csrfToken).withName("csrfToken");
+
+    ContainerTag languageForm = div();
+    if (profile.isPresent()) { // Show language switcher.
+      long userId = Long.parseLong(profile.get().getId());
+      String updateLanguageAction =
+          controllers.applicant.routes.ApplicantInformationController.updateWithRedirect(
+                  userId, request.uri())
+              .url();
+      String preferredLanguage = languageUtils.getPreferredLangage(request).orElse("en");
```

Comment:
suggestion (blocking): Use our default language

```suggestion
      String preferredLanguage = languageUtils.getPreferredLangage(request).orElse(LocalizedStrings.DEFAULT_LOCALE.toLanguageTag());
```

Reply:
If I use `MessagesApi` like you suggested above then this kind of takes care of itself, doesn't it?

Reply:
Yes, this looks nice!